Add Tiktok to OBW grow list

rjchow opened the pull request:
### All Submissions:

-   [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change?

<!-- Mark completed items with an [x] -->

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

This PR needs to land after 14609-gh-Automattic/woocommerce.com

### How to test the changes in this Pull Request:

With a new site,

1. Disable "Display suggestions within WooCommerce " in Settings > Advanced > WooCommerce.com.
2. Go through the Onboarding Wizard, select 'United States' as your location
3. Navigate to the Free Features section in the Onboarding Wizard.
4. See that TikTok is listed.
5. Select TikTok and complete the Onboarding Wizard.
6. On the WordPress Plugins page, confirm that TikTok for WooCommerce plugin is installed.

### Other information:

-   [x] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [ ] Have you written new tests for your changes, as applicable?
-   [x] Have you successfully run tests with your changes locally?
-   [x] Have you created a changelog file for each project being changed, ie `pnpm --filter=<project> run changelog add`?

<!-- Mark completed items with an [x] -->

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

rjchow committed:
add: tiktok to obw grow feed

chihsuan commented on plugins/woocommerce/src/Internal/Admin/RemoteFreeExtensions/DefaultFreeExtensions.php:
```diff
@@ -669,6 +670,221 @@ public static function get_plugin( $slug ) {
+				'name'           => __( 'TikTok for WooCommerce', 'woocommerce' ),
+				'image_url'      => plugins_url( '/assets/images/onboarding/tiktok.svg', WC_PLUGIN_FILE ),
+				'description'    =>
+					__( 'Create ad campaigns and reach one billion global users with TikTok for WooCommerce.', 'woocommerce' ),
```
Tested well! @rjchow  I'm wondering if we should add a link to "TikTok for WooCommerce" as well? 🤔 

<img src="https://user-images.githubusercontent.com/4344253/194201548-106a26c3-da9d-4d45-bec5-30ff16a6d6f3.png" width="500px" />

rjchow replied:
fixed! nice catch

rjchow committed:
removed geo restrictions and added hyperlink and name mapping

- geo restrictions not required for this promotion
- added name mapping for tiktok so that it shows up correctly in the obw plugin installation footer

rjchow committed:
added tiktok-for-business to valid plugin slugs

- OBW plugin installer page strips the :alt from the slug so we'll allow both variants

chihsuan reviewed (approved):
Thanks @rjchow looks great! 🚢
